File: Arrays/ThreeWayPartition/ThreeWayPartition.js
function method1(arr,low,high){
    let start=0, end=arr.length-1,temp,i;
    for(i=0;i<=end;){
        if(arr[i]<low){
            temp = arr[i];
            arr[i] = arr[start];
            arr[start] = temp;
            start++;
            i++;
        }
        else if(arr[i]>high){
            temp = arr[i];
            arr[i] = arr[end];
            arr[end] = temp;
            end--;
        }else{
            i++;
        }
    }
}


let arr = [1, 14, 5, 20, 4, 2, 54, 20, 87, 98, 3, 1, 32];
let low=10, high=20;
method1(arr, low, high);
console.log(arr);

File: Arrays/ThreeWayPartition/ThreeWayPartition.go
package main

import "fmt"

func method1(arr []int, a int, b int) {
	var i int
	var temp int
	n := len(arr)
	end := n - 1
	start := 0
	for i = 0; i <= end; {
		if arr[i] > b {
			temp = arr[i]
			arr[i] = arr[end]
			arr[end] = temp
			end--
		} else if arr[i] < a {
			temp = arr[i]
			arr[i] = arr[start]
			arr[start] = temp
			start++
			i++
		} else {
			i++
		}
	}
}

func main() {
	arr := []int{1, 14, 5, 20, 4, 2, 54, 20, 87, 98, 3, 1, 32}
	a, b := 10, 20
	method1(arr, a, b)
	fmt.Println(arr)
}
